ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -309,20 +327,56 @@ void ldmsd_store___del(ldmsd_cfgobj_t obj)
 int ldmsd_load_plugin(char* cfg_name, char *plugin_name,
 		      char *errstr, size_t errlen)
 {
+	const char *libpath;
 	struct ldmsd_plugin *api;
+	ldmsd_cfgobj_sampler_t sampler;
+	ldmsd_cfgobj_store_t store;
+	char log_name[1024];
+	ldmsd_cfgobj_t cfgobj;
+
 	if (!plugin_name || !cfg_name)
 		return EINVAL;
-	api = load_plugin(plugin_name);
+	/* If this plugin does not support multiple configurations,
+	 * the plugin_name must equal the cfg_name
+	 */
+	api = load_plugin(plugin_name, &libpath);
 	if (!api)
 		return errno;
+	if (0 == (api->flags & LDMSD_PLUGIN_MULTI_INSTANCE)) {
+		if (cfg_name && plugin_name) {
+			if (strcmp(cfg_name, plugin_name)) {
```

Comment:
I think we should do a lookup for existing use of the plugin rather than comparing cfg_name and plugin_name. There's really no reason that the configuration system needs to stop people from giving even a single plugin instance a name of their own creation. We'd need to document that exception in configuration syntax if we it it with strcmp here. Also this comment would be misleading when only a single configuration is being attempted.